src/thor/costmatrix.cc

```diff
@@ -896,29 +904,36 @@ void CostMatrix::CheckReverseConnections(const uint32_t target,
         uint32_t d = std::abs(static_cast<int>(rev_pred.path_distance()) +
                               static_cast<int>(fwd_label.path_distance()) -
                               static_cast<int>(fwd_label.transition_cost().secs));
-        best_connection_[source_idx].Update(fwd_edgeid, rev_pred.edgeid(), Cost(s, s), d);
-        best_connection_[source_idx].found = true;
+        best_connection_[idx].Update(fwd_edgeid, rev_pred.edgeid(), Cost(s, s), d);
```

Comment:
this is mostly changing `source_idx` name to `idx`; it's the connection's index, not the source index